Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs

```diff
@@ -0,0 +1,440 @@
+        {
+            var axes = new Vector3(quaternionArray[0], quaternionArray[1], quaternionArray[2]);
+
+            if (convert)
+            {
+                axes = Vector3.Scale(axes, CoordinateSpaceConversionScale) * -1.0f;
+            }
+
+            return new Quaternion(axes.x, axes.y, axes.z, quaternionArray[3]);
+        }
+
+        internal static float[] SetQuaternionValue(this Quaternion quaternion, bool convert = true)
+        {
+            // get the original axis and apply conversion scale as well as potential rotation axis flip
+            var axes = new Vector3(quaternion.x, quaternion.y, quaternion.z);
+
+            if (convert)
+            {
+                axes = Vector3.Scale(axes, CoordinateSpaceConversionScale) * 1.0f;
+            }
+
+            return new[] { axes.x, axes.y, axes.z, quaternion.w };
+        }
+
+        internal static Matrix4x4 GetMatrix4X4Value(this double[] matrixArray)
+        {
+            var matrix = new Matrix4x4(
+                new Vector4((float)matrixArray[0], (float)matrixArray[1], (float)matrixArray[2], (float)matrixArray[3]),
+                new Vector4((float)matrixArray[4], (float)matrixArray[5], (float)matrixArray[6], (float)matrixArray[7]),
+                new Vector4((float)matrixArray[8], (float)matrixArray[9], (float)matrixArray[10], (float)matrixArray[11]),
+                new Vector4((float)matrixArray[12], (float)matrixArray[13], (float)matrixArray[14], (float)matrixArray[15]));
+            Matrix4x4 convert = Matrix4x4.Scale(CoordinateSpaceConversionScale);
+            return convert * matrix * convert;
+        }
+
+        internal static float[] SetMatrix4X4Value(this Matrix4x4 matrix)
+        {
+            var convert = Matrix4x4.Scale(CoordinateSpaceConversionScale);
+            matrix = convert * matrix * convert;
+            return new[]
+            {
+                matrix.m00, matrix.m10, matrix.m20, matrix.m30,
+                matrix.m01, matrix.m11, matrix.m21, matrix.m31,
+                matrix.m02, matrix.m12, matrix.m22, matrix.m32,
+                matrix.m03, matrix.m13, matrix.m23, matrix.m33
+            };
+        }
+
+        internal static void GetTrsProperties(this Matrix4x4 matrix, out Vector3 position, out Quaternion rotation, out Vector3 scale)
+        {
+            position = matrix.GetColumn(3);
+
+            Vector3 x = matrix.GetColumn(0);
+            Vector3 y = matrix.GetColumn(1);
+            Vector3 z = matrix.GetColumn(2);
+
+            Vector3 calculatedZ = Vector3.Cross(x, y);
+            bool mirrored = Vector3.Dot(calculatedZ, z) < 0.0f;
+
+            scale.x = x.magnitude * (mirrored ? -1.0f : 1.0f);
+            scale.y = y.magnitude;
+            scale.z = z.magnitude;
+
+            rotation = Quaternion.LookRotation(matrix.GetColumn(2), matrix.GetColumn(1));
+        }
+
+        internal static int[] GetIntArray(this GltfAccessor accessor, bool flipFaces = true)
```

Comment:
>flipFaces [](start = 75, length = 9)

What is flipFaces?

Reply:
flip faces changes the triangle definitions/indices order. This changes the outward/normal direction which then results in changing which side of the mesh surface is drawn.

I think that the MRTK standard shader has some different flags that can override/disregard this behavior and draw both sides. But i would also guess this is a direct port from khronos's logic and should be kept